interpreter/program.go

```diff
@@ -102,14 +92,36 @@ func (p *exprProgram) GetInstruction(exprId int64) Instruction {
 	return p.instructions[p.revInstructions[exprId]]
 }
 
-func (p *exprProgram) Instructions() InstructionStepper {
+func (p *exprProgram) Init(dispatcher Dispatcher, state MutableEvalState) InstructionStepper {
+	if p.instructions == nil {
```

Comment:
Maybe call this Begin() ?

Reply:
Done, and I split the Init() step from the Begin() step.